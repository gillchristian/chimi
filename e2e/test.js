const fs = require('fs')
const path = require('path')

const shell = require('shelljs')

const chimiBin = path.resolve(__dirname, '..', 'bin', 'bin.js')
const fixturesRoot = path.join(__dirname, 'fixtures')

const fixturesDirs = fs
  .readdirSync(fixturesRoot)
  .map(x => path.join(fixturesRoot, x))
  .filter(x => fs.lstatSync(x).isDirectory())

describe('e2e tests', () => {
  for (let fixtureDir of fixturesDirs) {
    const relativeFixtureDir = path.relative(fixturesRoot, fixtureDir)
    it(`should run ${relativeFixtureDir} as expected`, () => {
      shell.cd(fixtureDir)

      const execution = shell.exec(`node ${chimiBin}`, { silent: true })
      const result = [execution.code, execution.stdout]

      expect(result).toMatchSnapshot()
    })
  }
})
