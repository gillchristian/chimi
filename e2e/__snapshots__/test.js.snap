// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e tests should run 01-simple-readme as expected 1`] = `
"
Output

----- README.md ----- 

--- Snippet #1 ---
42


 PASS  0-README.md

  Files: 1
  Snippets succeded: 1
"
`;

exports[`e2e tests should run 02-multiple-files as expected 1`] = `
"
Output

----- DOCS.md ----- 

--- Snippet #1 ---
42


----- README.md ----- 

--- Snippet #1 ---
42


 PASS  0-DOCS.md
 PASS  0-README.md

  Files: 2
  Snippets succeded: 2
"
`;
